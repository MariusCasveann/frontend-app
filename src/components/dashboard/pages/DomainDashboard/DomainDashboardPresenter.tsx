import * as React from 'react';
import { CardLayout } from '../../../common/molecules/CardLayout/CardLayout';
import DomainRessourcesPresenter from '../../molecules/DomainRessources/DomainRessourcesPresenter';
import CostPerDomainPresenter from '../../molecules/CostPerDomain/CostPerDomainPresenter';
import MonthlyCostPerDomainPresenter from '../../molecules/MonthlyCostPerDomain/MonthlyCostPerDomainPresenter';
import ProductCostPresenter from '../../molecules/ProductCost/ProductCostPresenter';
import '../common/Dashboard.css';
import DeploymentFrequencyPerDomainPresenter from '../../molecules/DeploymentFrequencyPerDomain/DeploymentFrequencyPerDomainPresenter';
import { DomainFiltersPresenter } from './DomainFiltersPresenter';
import { useState } from 'react';
import { QueryAllDomains_allDomains } from '../../atoms/DomainSelection/graphql/__generated__/QueryAllDomains';
import { FiltersContext } from '../../atoms/Filters/FiltersContext';
import { getCurrentDate, getYesterdayDate } from '../../../../utils/fiscalYear';

export default () => {
    const [selectedDomainState, setSelectedDomain] = useState<QueryAllDomains_allDomains | undefined>(undefined);
    const [fromDateState, setFromDate] = useState<string>(getYesterdayDate());
    const [toDateState, setToDate] = useState<string>(getCurrentDate());
    const [financialYearState, setFinancialYear] = useState<string | null>(null);

    return (
        <FiltersContext.Provider
            value={{
                fromDate: fromDateState,
                toDate: toDateState,
                setToDate,
                setFromDate,
                financialYear: financialYearState,
                setFinancialYear
            }}
        >
            <div className="dashboard-wrapper">
                <CardLayout width={100} className="dashboard-card-layout">
                    <DomainFiltersPresenter
                        selectedDomain={selectedDomainState}
                        setSelectedDomain={setSelectedDomain}
                    />
                    <DomainRessourcesPresenter selectedDomain={selectedDomainState} />
                    <ProductCostPresenter selectedDomain={selectedDomainState} />
                    <CardLayout width={100}>
                        <CostPerDomainPresenter selectedDomain={selectedDomainState} />
                        <DeploymentFrequencyPerDomainPresenter selectedDomain={selectedDomainState} />
                    </CardLayout>
                    <MonthlyCostPerDomainPresenter selectedDomain={selectedDomainState} />
                </CardLayout>
            </div>
        </FiltersContext.Provider>
    );
};
