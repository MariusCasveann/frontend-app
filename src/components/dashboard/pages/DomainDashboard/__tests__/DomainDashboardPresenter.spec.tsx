import * as React from 'react';
jest.mock('../../../../common/molecules/CardLayout/CardLayout', () => ({ Card: 'Card', CardLayout: 'CardLayout' }));
jest.mock('../../../molecules/DomainRessources/DomainRessourcesPresenter', () => 'DomainRessourcesPresenter');
jest.mock('../../../molecules/CostPerDomain/CostPerDomainPresenter', () => 'CostPerDomainPresenter');
jest.mock('../../../molecules/ProductCost/ProductCostPresenter', () => 'ProductCostPresenter');
jest.mock(
    '../../../molecules/MonthlyCostPerDomain/MonthlyCostPerDomainPresenter',
    () => 'MonthlyCostPerDomainPresenter'
);
jest.mock('../DomainFiltersPresenter', () => ({ DomainFiltersPresenter: 'DomainFiltersPresenter' }));
jest.doMock('../../../atoms/Filters/FiltersContext', () => ({
    DomainFiltersContext: 'FiltersContext',
    'DomainFiltersContext.Context': 'FiltersContext.Context'
}));

jest.doMock('../../../../../utils/fiscalYear', () => ({
    getCurrentDate: () => '2019-01-02',
    getYesterdayDate: () => '2019-01-01'
}));

import DomainDashboardPresenter from '../DomainDashboardPresenter';
import renderer from 'react-test-renderer';

describe(' DomainDashboardPresenter', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<DomainDashboardPresenter />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
