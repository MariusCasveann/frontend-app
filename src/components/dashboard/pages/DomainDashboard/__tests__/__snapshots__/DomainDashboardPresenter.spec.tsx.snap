// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` DomainDashboardPresenter renders correctly 1`] = `
<div
  className="dashboard-wrapper"
>
  <CardLayout
    className="dashboard-card-layout"
    width={100}
  >
    <DomainFiltersPresenter
      setSelectedDomain={[Function]}
    />
    <DomainRessourcesPresenter />
    <ProductCostPresenter />
    <CardLayout
      width={100}
    >
      <CostPerDomainPresenter />
      <Card
        className="deployment-frequency-per-resource-card"
        title="Deployment Frequency KPIs"
        width={45}
      />
    </CardLayout>
    <MonthlyCostPerDomainPresenter />
  </CardLayout>
</div>
`;
